go-controller/pkg/config/config_test.go

```diff
@@ -299,6 +331,100 @@ var _ = Describe("Config Operations", func() {
 		Expect(err).NotTo(HaveOccurred())
 	})
 
+	It("uses serviceaccount files", func() {
+		kubeconfigEnvFile, err := createTempFile("kubeconfig.env")
+		Expect(err).NotTo(HaveOccurred())
+		defer os.Remove(kubeconfigEnvFile)
+		os.Setenv("KUBECONFIG", kubeconfigEnvFile)
+		defer os.Setenv("KUBECONFIG", "")
+
+		kubeconfigFile, err := createTempFile("kubeconfig")
+		Expect(err).NotTo(HaveOccurred())
+		defer os.Remove(kubeconfigFile)
+
+		tokenFile, err1 := createTempFileContent("token", "TG9yZW0gaXBzdW0gZ")
+		Expect(err1).NotTo(HaveOccurred())
+		defer os.Remove(tokenFile)
+
+		kubeCAFile, err := createTempFile("ca.crt")
+		Expect(err).NotTo(HaveOccurred())
+		defer os.Remove(kubeCAFile)
+
+		cfgData := fmt.Sprintf(`[kubernetes]
+kubeconfig=%s
+apiserver=https://1.2.3.4:6443
+token=TG9yZW0gaXBzdW0gZ
+cacert=%s
+service-cidr=172.18.0.0/24`, kubeconfigFile, kubeCAFile)
+		err = ioutil.WriteFile(cfgFile.Name(), []byte(cfgData), 0644)
+		Expect(err).NotTo(HaveOccurred())
+
+		app.Action = func(ctx *cli.Context) error {
+			var cfgPath string
+			cfgPath, err = InitConfigSa(ctx, kexec.New(), tmpDir, nil)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(cfgPath).To(Equal(cfgFile.Name()))
+
+			Expect(Kubernetes.Kubeconfig).To(Equal(kubeconfigFile))
+			Expect(Kubernetes.CACert).To(Equal(kubeCAFile))
+			Expect(Kubernetes.Token).To(Equal("TG9yZW0gaXBzdW0gZ"))
+			Expect(Kubernetes.APIServer).To(Equal("https://1.2.3.4:6443"))
+			Expect(Kubernetes.ServiceCIDR).To(Equal("172.18.0.0/24"))
+
+			return nil
+		}
+		err = app.Run([]string{app.Name, "-config-file=" + cfgFile.Name()})
+		Expect(err).NotTo(HaveOccurred())
+
+	})
+
+	It("uses environment variables", func() {
+		kubeconfigEnvFile, err := createTempFile("kubeconfig.env")
+		Expect(err).NotTo(HaveOccurred())
+		defer os.Remove(kubeconfigEnvFile)
+		os.Setenv("KUBECONFIG", kubeconfigEnvFile)
+		defer os.Setenv("KUBECONFIG", "")
+
+		os.Setenv("K8S_TOKEN", "this is the  token test")
+		defer os.Setenv("K8S_TOKEN", "")
+
+		os.Setenv("K8S_APISERVER", "https://9.2.3.4:6443")
+		defer os.Setenv("K8S_APISERVER", "")
+
+		kubeCAFile, err1 := createTempFile("kube-ca.crt")
+		Expect(err1).NotTo(HaveOccurred())
+		defer os.Remove(kubeCAFile)
+		os.Setenv("K8S_CACERT", kubeCAFile)
+		defer os.Setenv("K8S_CACERT", "")
+
+		cfgData := fmt.Sprintf(`[kubernetes]
```

Comment:
Same here, the environment values should be different than the config file values, or really we don't pass a config file at all and make sure the environment values are correct.